mmdeploy/codebase/mmaction/deploy/video_recognition.py

```diff
@@ -0,0 +1,212 @@
+# Copyright (c) OpenMMLab. All rights reserved.
+from typing import Any, Dict, Optional, Sequence, Tuple, Union
+
+import mmengine
+import numpy as np
+import torch
+from mmengine.dataset import pseudo_collate
+from mmengine.model import BaseDataPreprocessor
+
+from mmdeploy.codebase.base import BaseTask
+from mmdeploy.utils import Task, get_root_logger
+from mmdeploy.utils.config_utils import get_input_shape
+from .mmaction import MMACTION_TASK
+
+
+def process_model_config(model_cfg: mmengine.Config,
+                         imgs: Union[Sequence[str], Sequence[np.ndarray]],
+                         input_shape: Optional[Sequence[int]] = None):
+    """Process the model config.
+
+    Args:
+        model_cfg (mmengine.Config): The model config.
+        imgs (Sequence[str] | Sequence[np.ndarray]): Input image(s), accepted
+            data type are List[str], List[np.ndarray].
+        input_shape (list[int]): A list of two integer in (width, height)
+            format specifying input shape. Default: None.
+
+    Returns:
+        mmengine.Config: the model config after processing.
+    """
+    logger = get_root_logger()
+    cfg = model_cfg.deepcopy()
+    test_pipeline_cfg = cfg.test_pipeline
+    if 'Init' not in test_pipeline_cfg[0]['type']:
+        test_pipeline_cfg = [dict(type='OpenCVInit')] + test_pipeline_cfg
+    else:
+        test_pipeline_cfg[0] = dict(type='OpenCVInit')
+    for i, trans in enumerate(test_pipeline_cfg):
+        if 'Decode' in trans['type']:
+            test_pipeline_cfg[i] = dict(type='OpenCVDecode')
+    cfg.test_pipeline = test_pipeline_cfg
+
+    # check whether input_shape is valid
+    if input_shape is not None:
+        has_crop = False
+        crop_size = -1
+        has_resize = False
+        scale = (-1, -1)
+        keep_ratio = True
+        for trans in cfg.test_pipeline:
+            if trans['type'] == 'Resize':
+                has_resize = True
+                keep_ratio = trans.get('keep_ratio', True)
+                scale = trans.scale
+            if trans['type'] in ['TenCrop', 'CenterCrop', 'ThreeCrop']:
+                has_crop = True
+                crop_size = trans.crop_size
+
+        if has_crop and tuple(input_shape) != (crop_size, crop_size):
+            logger.error(
+                f'`input shape` should be equal to `crop_size`: {crop_size},'
+                f' but given: {input_shape}')
+        if has_resize and (not has_crop):
+            if keep_ratio:
+                logger.error('Resize should set `keep_ratio` to False'
+                             ' when `input shape` is given.')
+            if tuple(input_shape) != scale:
+                logger.error(
+                    f'`input shape` should be equal to `scale`: {scale},'
+                    f' but given: {input_shape}')
+    return cfg
+
+
+@MMACTION_TASK.register_module(Task.VIDEO_RECOGNITION.value)
+class VideoRecognition(BaseTask):
```

Comment:
We need to add visualize method here since the visualize in BaseTask does not work on mmaction2.